src/activation.ts

```diff
@@ -132,36 +150,57 @@ function processActivatable(navigationInstruction: NavigationInstruction, callba
   return iterate();
 }
 
-function findActivatable(navigationInstruction: NavigationInstruction, callbackName: string, list: Array<Object> = [], router: Router): Array<Object> {
+interface IActivatableInfo {
+  viewModel: any;
+  lifecycleArgs: any[];
+  router: Router;
+}
+
+/**
+ * Find list of activatable view model and add to list (3rd parameter)
+ */
+function findActivatable(
+  navigationInstruction: NavigationInstruction,
+  callbackName: string,
+  list: IActivatableInfo[] = [],
+  router?: Router
+): IActivatableInfo[] {
   let plan = navigationInstruction.plan;
 
-  Object.keys(plan).filter((viewPortName) => {
-    let viewPortPlan = plan[viewPortName];
-    let viewPortInstruction = navigationInstruction.viewPortInstructions[viewPortName];
-    let viewModel = viewPortInstruction.component.viewModel;
-
-    if ((viewPortPlan.strategy === activationStrategy.invokeLifecycle || viewPortPlan.strategy === activationStrategy.replace) && callbackName in viewModel) {
-      list.push({
-        viewModel,
-        lifecycleArgs: viewPortInstruction.lifecycleArgs,
-        router
-      });
-    }
+  Object
+    .keys(plan)
+    .forEach((viewPortName) => {
+      let viewPortPlan = plan[viewPortName];
+      let viewPortInstruction = navigationInstruction.viewPortInstructions[viewPortName];
+      let viewModel = viewPortInstruction.component.viewModel;
+
+      if (
```

Comment:
nitpick but this is not any easier to read and takes up more vertical space, which i see as a net negative

Reply:
I see that it's not easier to read, but the current state doesn't seem to help either, so I picked the 80 chars a line rule and go with it. We can have this line as an exception though, what do you think about a tslint file